### DISCUSSION TITLE
Creating reusable responsive spacing utilities.

### DISCUSSION THREAD

**Discussion by dwkns** [tailwindlabs/tailwindcss#8195] [Help]
I have `<sections>` on a page where I want one of 3 paddings to apply — sm/md/lg to top/bottom or both of the element.

Each of these paddings have different sizes at various breakpoints. 

So I create in `main.css` ...

```css
/* Vertical Padding none/sm/md/lg — Top/Bottom/Both */
.pt-none {
  @apply pt-0 sm:pt-0 lg:pt-0

}
.pt-sm {
  @apply pt-2 sm:pt-2 lg:pt-2

}
.pt-md {
  @apply pt-4 sm:pt-8 lg:pt-12

}
.pt-lg {
  @apply pt-8 sm:pt-12 lg:pt-24
}

/* Bottom  */
.pb-none {
  @apply pb-0 sm:pb-0 lg:pb-0

}
.pb-sm {
  @apply pb-2 sm:pb-2 lg:pb-2

}
.pb-md {
  @apply pb-4 sm:pb-4 lg:pb-12

}
.pb-lg {
  @apply pb-8 sm:pb-8 lg:pb-24
}

/* Both  */
.py-none {
  @apply py-0 sm:py-0 lg:py-0

}
.py-sm {
  @apply pt-sm pb-sm

}
.py-md {
  @apply pt-md pb-md

}
.py-lg {
  @apply pt-lg pb-lg
}
```
This works but seems like a very long winded way of doing things. Especially if I want to create some margin classes in the same way. Is there a more elegant way of achieving these spacings  using the Tailwind Config file?

**Comment by bakura10** [comment]
If you have a set of common spacing on both top and bottom, the simplest is to copy-paste your code in the utilities layer. I am also doing that for some other use cases (for instance in our project we have a lot of scrollable areas where we use scroll-snap, overflow hidden, scroll mandatory) and that is defined as a shortcut like this.

```
.scroll-area {
   @apply overflow-x-auto overflow-y-hidden scrollbar-hidden snap-x snap-mandatory;
}
```

Your use case is pretty similar, but as it is very specific to your design and use cases, creating your own utility is definitely the way to go :)
